tests/e2e/e2e_setup_test.go

```diff
@@ -142,37 +147,40 @@ func (s *IntegrationTestSuite) SetupSuite() {
 	// 2. Start both networks.
 	// 3. Run IBC relayer betweeen the two chains.
 	// 4. Execute various e2e tests, including IBC.
+	var (
+		skipUpgrade bool
+		err         error
+	)
+
 	if str := os.Getenv("OSMOSIS_E2E_SKIP_UPGRADE"); len(str) > 0 {
-		skipUpgrade, err := strconv.ParseBool(str)
+		skipUpgrade, err = strconv.ParseBool(str)
 		s.Require().NoError(err)
+	}
 
-		if skipUpgrade {
-			debugRepository = "osmosis"
-			debugVersion = "debug"
-			s.configureDockerResources(chain.ChainAID, chain.ChainBID)
-			s.configureChain(chain.ChainAID, validatorConfigsChainA)
-			s.configureChain(chain.ChainBID, validatorConfigsChainB)
-
-			s.runValidators(s.chains[0], 0)
-			s.runValidators(s.chains[1], 10)
-			s.runIBCRelayer()
-			s.runPostUpgradeTests()
-			return
+	s.dockerImages = *dockerconfig.NewImageConfig(!skipUpgrade)
+
+	s.configureDockerResources(chain.ChainAID, chain.ChainBID)
+	s.configureChain(chain.ChainAID, validatorConfigsChainA, map[int]struct{}{
+		3: {}, // skip validator at index 3
+	})
+	s.configureChain(chain.ChainBID, validatorConfigsChainB, map[int]struct{}{})
+
+	for i, chainConfig := range s.chainConfigs {
+		s.runValidators(chainConfig, s.dockerImages.OsmosisRepository, s.dockerImages.OsmosisTag, i*10)
+	}
+
+	// Run a relayer between every possible pair of chains.
+	for i := 0; i < len(s.chainConfigs); i++ {
+		for j := i + 1; j < len(s.chainConfigs); j++ {
+			s.runIBCRelayer(s.chainConfigs[i].chain, s.chainConfigs[j].chain)
```

Comment:
I was thinking of still having the 2 chains but aiming to design for extensibility. So that if we want to add a third chain for any reason, like testing some ibc functionality in the future, it works out of the box.

The primary reason for this refactor though is to just encapsulate more "chain-specific" data under the same abstraction. That allows me to add more methods on that abstraction related to state-sync. This particular change is secondary and doesn't add more value to the end goal